pkg/api/norman/store/app/store.go

```diff
@@ -57,11 +64,16 @@ func (s *Store) Update(apiContext *types.APIContext, schema *types.Schema, data
 		return nil, err
 	}
 
-	if err := s.validateRancherVersion(data); err != nil {
+	if err := s.validateForMultiClusterApp(id, "update"); err != nil {
 		return nil, err
 	}
 
-	if err := s.validateForMultiClusterApp(id, "update"); err != nil {
+	clusterName, err := s.parseClusterID(apiContext.Request.URL.Path)
```

Comment:
nit: var name is clusterName, but method is parseClusterID. not super familiar with this area, so my bad if those are commonly interchangeable.

Reply:
They are but I change it anyway